src/postgres/getClientConfig.ts

```diff
@@ -17,21 +17,36 @@ export async function getClientConfig(treeItem: PostgresServerTreeItem, database
 
     if (!(username && password)) {
         const credentials = await treeItem.getCredentials();
-        username = nonNullProp(credentials, 'username');
-        password = nonNullProp(credentials, 'password');
+        if (credentials.username && credentials.password) {
+            username = credentials.username;
+            password = credentials.password;
+        }
     }
 
-    const ssl: ConnectionOptions = {
+    const sslAzure: ConnectionOptions = {
         // Always provide the certificate since it is accepted even when SSL is disabled
         // Certificate source: https://aka.ms/AA7wnvl
         ca: BaltimoreCyberTrustRoot
     };
 
-    if (username && password) {
+    if ((username && password) || username === 'postgres') {
         const host = nonNullProp(treeItem.connectionString, 'hostName');
         const port: number = treeItem.connectionString.port ? parseInt(treeItem.connectionString.port) : postgresDefaultPort;
-        const clientConfig = { user: username, password: password, ssl, host, port, database: databaseName };
-        const client = new Client(clientConfig);
+        let ssl: boolean | ConnectionOptions | undefined;
+        if (treeItem.azureName) {
+            ssl = sslAzure;
+        } else {
+            ssl = treeItem.connectionString.ssl;
+        }
+        let clientConfig: ClientConfig;
+        let client: Client;
+        if (treeItem.azureName) {
```

Comment:
Even though I'm fine with you not doing `createPgClient` for this PR, I still want you to take the other part of that suggestion, which more clearly separated out the azure vs. attached scenarios. Aka something like this:
```typescript
export async function getClientConfig(treeItem: PostgresServerTreeItem, databaseName: string): Promise<ClientConfig> {
    let clientConfig: ClientConfig;
    if (treeItem.azureName) {
        // all the stuff to create clientConfig for the Azure case
    } else {
        clientConfig = { connectionString: treeItem.connectionString.connectionString };
    }

    // Ensure the client config is valid before returning
    const client = new Client(clientConfig);
    try {
        await client.connect();
        return clientConfig;
    } finally {
        await client.end();
    }
}
```